src/ssa/local_ssa.cpp

```diff
@@ -481,7 +481,12 @@ void local_SSAt::build_transfer(locationt loc)
       }
       else
       {
-        assign_rec(deref_lhs, rhs, true_exprt(), loc);
+        assign_rec(symbolic_deref_rhs, deref_rhs, true_exprt(), loc);
+        assign_rec(
+          deref_lhs,
+          name(ssa_objectt(symbolic_deref_rhs, ns), OUT, loc),
+          true_exprt(),
+          loc);
```

Comment:
This should be placed outside the if statement since it does not cover the situation: p->next = q->next (when there is symbolic dereference at LHS).